### ISSUE TITLE
XAML hot-reload doesn't work anymore

### ISSUE THREAD

**Issue by artkat** [microsoft/microsoft-ui-xaml#5324]
**Describe the bug**
Most changes to XAML while debugging an application would visible immediately in the application. This is not the case anymore after upgrading to v0.8.0 

**Steps to reproduce the bug**
1. Start a new WinUI solution in Visual Studio
2. Debug the application
3. Open MainWindow.XAML
4. Change the background color of the root item to red

**Expected behavior**
Expected to see the change (red control)

**Version Info**
WinUI 0.8.0
Visual Studio 2019 v16.10.3

NuGet package version: 

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
|  |  |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| October 2020 Update (19042)        |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            |  |

**Additional context**
Since WinUI doesn't come with a UI designer, it's crucial hot-reload of XAML works. Recompile/start between every XAML change to see the result is too time-consuming.

**Comment by madakr333** [comment]
After researching Desktop frameworks, found out that WinUI is the most prominent one and recommended to use for new projects. That was a huge mistake. I have already spent 3 days just to set up the solution. Many bugs, which should have been fixed even in earlier versions, like extending a title bar still doesn't work in WinUI 3, hot reloading doesn't work at all, building almost empty project takes 1 minute(and I have modern laptop). Idk what you're guys doing there. It all just looks good for presentation, but in production it is buggy piece of crap.

**Comment by StephenLPeters** [comment]
@fabiant3 FYI

**Comment by pablojimpas** [comment]
Has anyone found a workaround to this issue?

**Comment by fabiant3** [comment]
@pablojimpas - You can find more info here: [Release Notes](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#known-issues )

WinUI 3 tooling such as Live Visual Tree, Live Property Explorer, and Hot Reload are currently not supported for the 0.8 release. You can expect to see support for these features in Visual Studio 2019 16.11 Preview 3.

**Comment by alwu-msft** [comment]
@artkat Were you able to verify that [Visual Studio 2019 16.11 Preview 3](https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-preview#16.11.0-pre.3.0) supports XAML Hot Reload?

**Comment by harvinders** [comment]
I am not able to use XAML hot reload in 16.11. I think there was some preview feature that we need to enable to get it working, but I am not seeing the option.

**Comment by RealTommyKlein** [comment]
@harvinders - that preview feature is removed and no longer needed to enable Hot Reload.  Do you see an error message in the Live Visual Tree window (`Debug -> Windows -> Live Visual Tree` when the app is running), and are you on Reunion 0.8 or later which are the supported versions in 16.11?

**Comment by harvinders** [comment]
@RealTommyKlein I do see the `Live Visual Tree`. It's just that the XAML hot reload is not working.

Yes, I am using 0.8 release and the repo can be found [here](https://github.com/harvinders/CommandBinding)

![image](https://user-images.githubusercontent.com/8557046/129732031-3d8d03cf-d223-453a-a82d-d8e3cf2b8be1.png)

![image](https://user-images.githubusercontent.com/8557046/129732333-b3797ee8-91f6-4ff6-82b3-becb2302243c.png)

**Comment by RealTommyKlein** [comment]
@harvinders - thanks, can you describe what Hot Reload operation you're making that isn't working?  E.g. I tried adding a `TextBlock` to the app's `MainPage.xaml` and saw it correctly reflected in the app:

![hotreload](https://user-images.githubusercontent.com/45502154/129751948-b7711784-2caa-4416-812c-bec71371579c.png)

**Comment by harvinders** [comment]
@RealTommyKlein Nothing seems to be working, I tried adding a `Background` property to say `Grid` or changing the `Text` property of a `TextBlock`.

**Comment by RealTommyKlein** [comment]
@harvinders unfortunately I still can't repro the issue :unamused:.  I tried the `Grid.Background` edit on the `ContentArea` `Grid` in `MainPage.xaml`,  and the `TextBlock.Text` edits in both `MainPage.xaml` and `NotificationView.xaml` in 16.11.1 and saw them render correctly.  As a sanity-check, can you try saving the `.xaml` file after making the changes in the running app and see if they render correctly (there's an option that may be enabled, `Tools -> Options -> Debugging -> Hot Reload -> Apply XAML Hot Reload on document save`, which causes Hot Reload changes to only apply when the markup file is saved instead of applying instantly).  If that still doesn't work, can you describe the exact edit you're making (which page you're modifying, the line number of the element, its property you're modifying, and the new property value).

It also worked in my testing on VS2022, although the XAML Live Preview window you posted is a new feature which is different than the XAML Live Visual Tree window:
![dev17](https://user-images.githubusercontent.com/45502154/130122580-58da944f-bfbb-4eb3-95b7-c535025df091.png)

Can you sanity check that WinUI Hot Reload is enabled under `Tools -> Options -> Debugging -> XAML Hot Reload -> WinUI (including .NET MAUI)`)?  Thanks!

**Comment by harvinders** [comment]
@RealTommyKlein So these are things that I have tried

1. Add a new `<TextBlock Text="testing"/>` at the same level as yours
2. I also tried changing the background of `ContentArea` 
3. Also tried the settings in the options, but without success

4. I did not get the time to test your suggestion with VS2022, will try tomorrow morning.

However, it is not showing up. I thought that my visual studio installation may not be proper, so I uninstalled and reinstalled it, but without any success. I am using Windows 10, 1909. Could that be the reason? I am going to attempt and try it on another PC tomorrow or day after and see what happens.

**Comment by akilarajesh1313** [comment]
@harvinders Have you tried with VS2022 and/or another PC yet? I'm not sure what the source of the issue is but our teams are currently investigating. Thank you for your patience!

**Comment by harvinders** [comment]
@akilarajesh1313 I have now tried with visual studio 2022 also, but no luck, hot reload is still not working.

**Comment by RealTommyKlein** [comment]
Thanks @harvinders, was finally able to repro it!  Looks like an OS-related bug.  On Windows 1909 (18363) I ran into the issue, but not in an identical setup on future OS versions (e.g. 19043 worked).  As a workaround, are you able to upgrade your OS version?  Either version of VS (16.11 or VS2022) with the workloads you described should work after updating the OS.

**Comment by harvinders** [comment]
Unfortunately, we can't upgrade the OS, these are company issued laptops, and they are likely to update only after the support for 1909 goes away, which is probably still one year away.

**Comment by harvinders** [comment]
@RealTommyKlein, It seems I can upgrade to 20H1, do you know if hot reload works on 20H1? Also, I believe Microsoft would fix the issue and not ask everyone to just upgrade the OS.

**Comment by RealTommyKlein** [comment]
@harvinders - yes, tested and Hot Reload will work on 20H1.  We plan to fix the issue, but upgrading the OS is an immediate workaround.  The most likely culprit is Visual Studio or Windows App SDK code that depends on the OS version, meaning it would be 1-3 months before a non-preview release of those is available with the fix.  Additionally, if the Visual Studio/Windows App SDK versions are controlled by your organization like the OS version is, it could be even longer before you'd be able to use the fix.  The last but unlikely case is a bug in the OS itself, which wouldn't meet the bar for an OS servicing release and would never be fixed.

**Comment by harvinders** [comment]
@RealTommyKlein, I finally managed to get my OS upgraded. I can also confirm that it works on 20H2. It is such a relief, I never thought much about this feature and took it for granted, but lack of it completely crippled the productivity. 

Thanks

**Comment by artkat** [comment]
**XAML hot-reload is broken again**

Visual Studio 2022 Version 17.0.0 Preview 5.0 
WinUI  0.8.5
Windows 11
